lib/jobs.js

```diff
@@ -61,8 +61,21 @@ function start(buildConfig, callback) {
  * @param {Function}   callback                   Callback fn(error, result)
  */
 function stop(buildConfig, callback) {
-    return redis.hdel(`${queuePrefix}buildConfigs`, buildConfig.buildId)
-        .then(() => executor.stop(buildConfig))
+    const stopConfig = { buildId: buildConfig.buildId };
+
+    return redis.hget(`${queuePrefix}buildConfigs`, buildConfig.buildId)
+        .then((fullBuildConfig) => {
+            const parsedConfig = JSON.parse(fullBuildConfig);
+
+            if (parsedConfig && parsedConfig.annotations) {
+                stopConfig.annotations = parsedConfig.annotations;
```

Comment:
You might want to use `parsedConfig` here like `start` function. Or use `stopConfig` the case `parsedConfig` is empty. I think it is not good idea to overwrite only `annotations` key directly.
```
parsedConfig = parsedConfig || stopConfig;
```

Reply:
Only `buildId` and `annotations` are permitted in stop schema.
https://github.com/screwdriver-cd/data-schema/blob/3df5ebc/plugins/executor.js#L16-L19

To pass a validation, I just overwrite `annotations` directly.

Reply:
🙆